playbooks/robusta_playbooks/k8s_resource_enrichments.py

```diff
@@ -43,32 +72,105 @@ def to_pod_row(pod: Pod, cluster_name: str) -> List:
         pods = job_pods if job_pods else []
-    elif resource.kind == "Pod":
+    elif kind == "Pod":
         pods = [resource]
-    elif resource.kind == "Node":
+    elif kind == "Node":
         pods = Pod.listPodForAllNamespaces(field_selector=f"spec.nodeName={resource.metadata.name}").obj.items
     else:
         selector = build_selector_query(resource.spec.selector)
         pods = PodList.listNamespacedPod(namespace=resource.metadata.namespace, label_selector=selector).obj.items
 
+    return pods
+
+def to_pod_obj(pod: Pod, cluster: str) -> RelatedPod:
+    resource_requests = pod_requests(pod)
+    resource_limits = pod_limits(pod)
+    addresses = ",".join([address.ip for address in pod.status.podIPs])
+    return RelatedPod(
+        name=pod.metadata.name,
+        namespace=pod.metadata.namespace,
+        node=pod.spec.nodeName,
+        clusterName=cluster,
+        cpuLimit=resource_limits.cpu,
+        cpuRequest=resource_requests.cpu,
+        memoryLimit=resource_limits.memory,
+        memoryRequest=resource_requests.memory,
+        creationTime=pod.metadata.creationTimestamp,
+        restarts=pod_restarts(pod),
+        addresses=addresses,
+        containers=get_pod_containers(pod),
+        status=pod.status.phase,
+    )
+
+def get_pod_containers(pod: Pod) -> List[RelatedContainer]:
+    containers: List[RelatedContainer] = []
+    for container in pod.spec.containers:
+        requests = PodContainer.get_requests(container)
+        limits = PodContainer.get_limits(container)
+        containerStatus: Optional[ContainerStatus] = PodContainer.get_status(pod, container.name)  
+        currentState : Optional[ContainerState] = getattr(containerStatus, "state", None)
+        stateStr : str = "waiting"
+        state = None
+        if currentState:
+            for s in ["running", "waiting" , "terminated"]:
+                state = getattr(currentState, s, None)
+                if state is not None:
+                    stateStr = s
+                    break
+
+        containers.append(
+            RelatedContainer(
+            name=container.name,
+            cpuLimit=limits.cpu,
+            cpuRequest=requests.cpu,
+            memoryLimit=limits.memory,
+            memoryRequest=requests.memory,
+            restarts=getattr(containerStatus, "restartCount", 0),
+            status=stateStr,
+            created=getattr(state, "startedAt", None)
+            )
+        )
+    
+    return containers
+ 
+
+
+
+@action
+def resource_related_pods(event: KubernetesResourceEvent):
+    """
+    Return the list of pods and containers related to that k8s resource.
+    For example, return all pods of a given k8s Deployment
+
+    Supports Deployments, ReplicaSets, DaemonSets, StatefulSets and Pods
+    """
+    pods = get_related_pods(event.get_resource())
+    cluster = event.get_context().cluster_name
+    event.add_enrichment([
+       JsonBlock(json.dumps([to_pod_obj(pod, cluster).dict() for pod in pods]))
```

Comment:
I don't understand why you're using json blocks, that can only be used by one sink

Reply:
In addition, this will work only on newer runner versions.
Why would we want that?